plugin/kubernetes/controller.go

```diff
@@ -21,13 +22,18 @@ import (
 const (
 	podIPIndex            = "PodIP"
 	svcNameNamespaceIndex = "NameNamespace"
+	ingNameNamespaceIndex = "IngressNameNamespace"
 	svcIPIndex            = "ServiceIP"
+	ingHostIndex          = "IngressHost"
 	epNameNamespaceIndex  = "EndpointNameNamespace"
 	epIPIndex             = "EndpointsIP"
 )
 
 type dnsController interface {
 	ServiceList() []*object.Service
+	IngressList() []*object.Ingress
+	IngIndex(string) []*object.Ingress
+	IngIndexReverse(string) []*object.Ingress
```

Comment:
This isn't ideal - that we have to modify the controller interface like this.
Perhaps there is another approach...